tiledb/api/src/group/mod.rs

```diff
@@ -0,0 +1,745 @@
+            raw: RawTDBString::Owned(tiledb_uri),
+        }
+        .to_string();
+        let uri = uri?;
+
+        let object_type = ObjectType::try_from(tiledb_type)?;
+        Ok(GroupInfo {
+            uri,
+            group_type: object_type,
+            name,
+        })
+    }
+
+    pub fn is_relative_uri<S>(&self, name: S) -> TileDBResult<bool>
+    where
+        S: AsRef<str>,
+    {
+        let c_context = self.context.capi();
+        let mut c_relative: u8 = out_ptr!();
+        let c_name = cstring!(name.as_ref());
+        self.context.capi_return(unsafe {
+            ffi::tiledb_group_get_is_relative_uri_by_name(
+                c_context,
+                Self::capi(self),
+                c_name.as_ptr(),
+                &mut c_relative,
+            )
+        })?;
+        Ok(c_relative > 0)
+    }
+
+    pub fn is_open(&self) -> TileDBResult<bool> {
+        let c_context = self.context.capi();
+        let mut c_open: i32 = out_ptr!();
+        self.context.capi_return(unsafe {
+            ffi::tiledb_group_is_open(c_context, Self::capi(self), &mut c_open)
+        })?;
+        Ok(c_open > 0)
+    }
+
+    pub fn dump(&self, recursive: bool) -> TileDBResult<Option<String>> {
+        let c_context = self.context.capi();
+        let mut c_str: *mut std::ffi::c_char = out_ptr!();
+        let c_recursive = if recursive { 1 } else { 0 };
+        self.context.capi_return(unsafe {
+            ffi::tiledb_group_dump_str(
+                c_context,
+                Self::capi(self),
+                &mut c_str as *mut *mut std::ffi::c_char,
+                c_recursive,
+            )
+        })?;
+        let group_dump = unsafe { std::ffi::CStr::from_ptr(c_str) };
+        let group_dump_rust_str = group_dump.to_string_lossy().into_owned();
+
+        Ok(Some(group_dump_rust_str))
+    }
+
+    pub fn set_config(&mut self, config: &Config) -> TileDBResult<()> {
+        let c_context = self.context.capi();
+        let cfg = config.capi();
+        self.context.capi_return(unsafe {
+            ffi::tiledb_group_set_config(c_context, Self::capi(self), cfg)
+        })?;
+        Ok(())
+    }
+
+    pub fn config(&self) -> TileDBResult<Config> {
+        let c_context = self.context.capi();
+        let mut c_cfg: *mut ffi::tiledb_config_t = out_ptr!();
+        self.context.capi_return(unsafe {
+            ffi::tiledb_group_get_config(
+                c_context,
+                Self::capi(self),
+                &mut c_cfg,
+            )
+        })?;
+
+        Ok(Config {
+            raw: RawConfig::Owned(c_cfg),
+        })
+    }
+
+    pub fn put_metadata(&mut self, metadata: Metadata) -> TileDBResult<()> {
+        println!("{:?}", metadata);
+        let c_context = self.context.capi();
+        let (vec_size, vec_ptr, datatype) = metadata.c_data();
+        println!("{:?}", metadata);
+        let c_key = cstring!(metadata.key.clone()); // we're partially moving metadata
```

Comment:
Once you remove the `println`s I think you probably won't need the `clone` any more